packages/wordpress-kit/src/lib/tailwindcssPlugin/utilities/Colors.ts

```diff
@@ -0,0 +1,214 @@
+ * @class ColorUtilities
+ * @param {ThemeType} theme - The theme object from tailwindcss plugin
+ *
+ * @example
+ * ```
+ * const color = new ColorUtilities(theme);
+ *
+ * color.getBackgroundUtilities() // returns the background utilities
+ * color.getBorderColorUtilities() // returns the border color utilities
+ * ```
+ *
+ */
+export class ColorUtilities {
+  private theme: ThemeType;
+
+  constructor(theme: ThemeType) {
+    this.theme = theme;
+  }
+
+  /**
+   * Checks if the color is in the theme colors list
+   * is a tailwind default color. If not, returns a hex color.
+   *
+   * @param color - A color object
+   * @returns string - A string with the color value
+   */
+  private getColor = (color: Color) =>
+    this.theme(
+      `colors.${color.themeName}`,
+      this.theme(`colors.${color.tailwindDefault}`, color.hexDefault)
+    );
+
+  /**
+   * Finds a color by themeName
+   *
+   * @param {string} colorName
+   * @return Color
+   * @throws Error(`Color ${colorName} not found`)
+   */
+  private findColor = (colorName: string) => {
+    const color = colorList.find(color => color.themeName === colorName);
+    if (!color) {
+      throw new Error(`Color ${colorName} not found`);
+    }
+    return color;
+  };
+
+  /**
+   * @private
+   * Formats color gradients from a list of colors.
+   *
+   * @remarks
+   * to be used in the linear-gradient css function.
+   * @param {GradientColors[]} colors - Array of gradient colors i.e
+   *
+   * ```
+   * [{ color: 'primary', position: '0%' }, ...]
+   * ```
+   *
+   * @returns {string} A string to be used as a css gradient. Example: '#000000 50%, #ffffff 100%'
+   */
+  private formatGradientColors = (colors: GradientColors[]) => {
+    const formattedColors = colors.map(({ color, position }) => {
+      if (color.startsWith('#')) {
+        return `${color} ${position}`;
+      }
+
+      const themeColor = this.findColor(color);
+      return `${this.getColor(themeColor)} ${position}`;
+    });
+
+    return formattedColors.join(', ');
+  };
+
+  /**
+   *
+   * Takes the colors from the colorList constant and
+   * build an object with the following structure:
+   *
+   * @returns ```{[`.has-${colorName}-color`]: { color: 'colorValue' }, ... }``` - color utilities.
+   *
+   * @example
+   * const colorUtilities = getColorUtilities();
+   * ```
+   * {
+   *  '.has-primary-color': {
+   *   color: '#0070f3',
+   *  },
+   *  ...
+   * },
+   * ```
+   */
+  getColorUtilities = () =>
+    colorList.reduce(
+      (acc, color) => ({
+        ...acc,
+        [`.has-${color.name}-color`]: {
+          color: this.getColor(color),
+        },
+      }),
+      {}
+    );
+
+  /**
+   *
+   * Takes the colors from the colorList constant to
+   * build an object with the background colors and a padding.
+   *
+   * @returns ```{[`.has-${colorName}-background-color`]: { backgroundColor: 'colorValue' }, ... }``` - background utilities.
```

Comment:
If you format the triple backticks, it will show the proper code formatting in the TSDoc (at least it does in VSCode)

```suggestion
   * @returns ```
   * {[`.has-${colorName}-background-color`]: {
   *   backgroundColor: 'colorValue' },
   *   ...
   * }
   * ```
```